http-client-core/src/main/java/io/micronaut/http/client/exceptions/HttpClientResponseException.java

```diff
@@ -83,11 +83,27 @@ public HttpResponse<?> getResponse() {
 
     /**
      * @return The {@link io.micronaut.http.HttpStatus} returned
+     * @deprecated To support custom status codes. Use {@link #code()} instead of {@link #getStatus()} and {@link HttpStatus#getCode()}  and {@link #reason()} instead of {@link #getStatus()} and {@link HttpStatus#getReason()}
```

Comment:
```suggestion
     * To support custom status codes. Use {@link #code()} instead of {@link #getStatus()} and {@link HttpStatus#getCode()}  and {@link #reason()} instead of {@link #getStatus()} and {@link HttpStatus#getReason()}.
```